File: mRemoteNG/UI/UI.Window.Base.vb
Imports WeifenLuo.WinFormsUI.Docking

Namespace UI
    Namespace Window
        Public Class Base
            Inherits DockContent

#Region "Public Properties"
            Private _WindowType As UI.Window.Type
            Public Property WindowType() As UI.Window.Type
                Get
                    Return Me._WindowType
                End Get
                Set(ByVal value As UI.Window.Type)
                    Me._WindowType = value
                End Set
            End Property

            Private _DockPnl As DockContent
            Public Property DockPnl() As DockContent
                Get
                    Return Me._DockPnl
                End Get
                Set(ByVal value As DockContent)
                    Me._DockPnl = value
                End Set
            End Property
#End Region

#Region "Public Methods"
            Public Sub SetFormText(ByVal Text As String)
                Me.Text = Text
                Me.TabText = Text
            End Sub
#End Region

#Region "Private Methods"
            Private Sub Base_Load(sender As System.Object, e As System.EventArgs) Handles MyBase.Load
                frmMain.ShowHidePanelTabs()
            End Sub

            Private Sub Base_FormClosed(sender As System.Object, e As System.Windows.Forms.FormClosedEventArgs) Handles MyBase.FormClosed
                frmMain.ShowHidePanelTabs(Me)
            End Sub
#End Region
        End Class
    End Namespace
End Namespace

File: mRemoteV1/UI/Windows/UI.Window.Base.cs
using System.Windows.Forms;
using WeifenLuo.WinFormsUI.Docking;


namespace mRemoteNG.UI.Window
{
	public class Base : DockContent
    {
        #region Private Variables
        private Type _WindowType;
        private DockContent _DockPnl;
        #endregion

        #region Constructors
        public Base()
		{
			//InitializeComponent();
		}
        #endregion

        #region Public Properties
        public Type WindowType
		{
			get { return this._WindowType; }
			set { this._WindowType = value; }
		}
		
        public DockContent DockPnl
		{
			get { return this._DockPnl; }
			set { this._DockPnl = value; }
		}
        #endregion
				
        #region Public Methods
		public void SetFormText(string Text)
		{
			this.Text = Text;
			this.TabText = Text;
		}
        #endregion
				
        #region Private Methods
		private void Base_Load(System.Object sender, System.EventArgs e)
		{
			frmMain.Default.ShowHidePanelTabs();
		}
				
		private void Base_FormClosed(System.Object sender, System.Windows.Forms.FormClosedEventArgs e)
		{
			frmMain.Default.ShowHidePanelTabs(this);
		}
        #endregion
	}
}